ouroboros-consensus/docs/diffusion-pipelining-design/block-diffusion-pipelining.tex

```diff
@@ -0,0 +1,554 @@
+  \item If the last block slot numbers are the same then order by whether \emph{this} node produced the block or not (locally produced blocks preferred);
+  \item If the last blocks were issued by the same entity (same SPO cold key), then order by the hot key certificate issue number (higher cert issue number preferred);
+  \item Finally, order by VRF value (lower VRF value preferred).
+\end{enumerate}
+
+An appropriate way to analyse this is to argue that there can only be bounded
+progress in this order.
+
+\begin{enumerate}
+  \item In the short term, the chain length is bounded by the Ouroboros leader schedule.
+  \item The second condition is boolean, so it is bounded.
+  \item The certificate counters are controlled by the block issuers and are not inherently bounded. This needs special treatment to establish a bound.
+  \item The VRF value depends on the identity of the block issuer, but its value is not under the issuer's control. We rely on the property of the VRF primitive that it's only possible to create a single valid value per block issuer, per slot. Given this, the number of different VRF values is bounded by the number of different authorised block issuers (from the end of the most recent common chain) which is then bounded by the Ouroboros leader schedule.
+\end{enumerate}
+
+\emph{So the only case that needs special treatment is bounding the certificate issue numbers.} \\
+
+The purpose of the certificate issue numbers is to enable a block producer to
+cycle their hot keys and in particular to be able to issue a new hot key if the
+current hot key is compromised. The reason one might compare on certificate issue
+numbers at all is in the situation where a compromised hot key is being used
+to sign blocks. In this situation the true owner needs to issue a new hot key
+with a new, higher certificate issue number, via their cold key, and then 
+use that new hot key to sign blocks. 
+
+This results in the possibility that the stolen key and the new key (with corresponding certs) 
+may be being used at the same time, and they have to compete to be selected. This is why the 
+chain selection rule prefers the later issue number, so it will be selected even if the block 
+signed by the compromised key arrives first and is thus selected first.
+
+In principle the certificate issue numbers are not bounded, but the use case
+of recovering from compromised keys does not require more than two issue
+numbers. So 'progress' in the certificate issue number can be bounded by
+allowing at most two different blocks with the same issuer in the same slot.
+There is also a 'no regression' rule with regard to the issue numbers used on the chain, so
+even if an attacker has several compromised certificates, as soon as they
+use a later one, then the earlier ones can no longer be used again.
+
+\subsection{Many valid block bodies}
+The case for many block bodies is straightforward: they make no difference to
```

Comment:
I think this section should explicitly emphasize that different blocks have different headers, so the previous subsection already addresses every pertinent concern re DoS.